File: articles-api-typescript/src/entities/Tag.ts
import "reflect-metadata";
import { Entity, PrimaryColumn, Column } from "typeorm";
import { v4 as uuidv4 } from "uuid";

@Entity("tags")
export class Tag {
  @PrimaryColumn("varchar")
  id: string;

  @Column("varchar")
  name: string;

  constructor(name: string) {
    this.id = uuidv4();
    this.name = name ?? "";
  }
}

File: article-service/src/main/java/io/spring/articleservice/model/Tag.java
package io.spring.articleservice.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.util.UUID;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "tags")
@Getter
@Setter
@NoArgsConstructor
public class Tag {

  @Id
  @Column(length = 255)
  private String id;

  @Column(nullable = false, length = 255, unique = true)
  private String name;

  public Tag(String name) {
    this.id = UUID.randomUUID().toString();
    this.name = name;
  }
}
